Prox/Prox/PlaceDetails/PlaceDetailViewController.swift

```diff
@@ -97,52 +110,242 @@ class PlaceDetailViewController: UIViewController {
         // Dispose of any resources that can be recreated.
     }
 
+    fileprivate func dequeuePlaceCardViewController(forPlace place: Place) -> PlaceDetailsCardViewController {
+        if let placeCardVC = unusedPlaceCardViewControllers.last {
+            unusedPlaceCardViewControllers.removeLast()
+            placeCardVC.place = place
+            return placeCardVC
+        }
 
-    func pageControlDidPage(sender: AnyObject) {
-        let pageSize = imageCarousel.bounds.size
-        let xOffset = pageSize.width * CGFloat(pageControl.currentPage)
-        imageCarousel.setContentOffset(CGPoint(x: xOffset, y: 0), animated: true)
+        return PlaceDetailsCardViewController(place: place)
     }
-    
-}
 
-extension PlaceDetailViewController: UICollectionViewDataSource {
-    func numberOfSections(in collectionView: UICollectionView) -> Int {
-        return 1
+    var startConstant: CGFloat!
+
+    func didPan(gestureRecognizer: UIPanGestureRecognizer) {
+
+        if gestureRecognizer.state == .began {
+            startConstant = currentCardViewCenterXConstraint?.constant ?? 0
+        }
+
+        let translationX = gestureRecognizer.translation(in: self.view).x
+
+        if gestureRecognizer.state == .ended {
+            // figure out where the view would stop based on the velocity with which the user is panning
+            // this is so that paging quickly feels natural
+            let velocityX = (0.2 * gestureRecognizer.velocity(in: self.view).x)
+            let finalX = startConstant + translationX + velocityX;
```

Comment:
Shouldn't velocity be multiplied by the unit of time? In this case, `animationDuration`?